### PR TITLE
Cleanup umul()

### PR DESCRIPTION
**Pull request by gumb0**
Pulled out of https://github.com/chfast/intx/pull/180 with a test fixed in a boring way.

### CODE REVIEW

**Comment by gumb0** [comment]
@chfast Not sure what to do with coverage here, any ideas?

File: include/intx/int128.hpp [hunk 1]
```diff
@@ -358,9 +358,23 @@ constexpr uint128 operator>>(uint128 x, uint128 shift) noexcept
...
-/// Portable full unsigned multiplication 64 x 64 -> 128.
-constexpr uint128 constexpr_umul(uint64_t x, uint64_t y) noexcept
+
+/// Full unsigned multiplication 64 x 64 -> 128.
+inline constexpr uint128 umul(uint64_t x, uint64_t y) noexcept
 {
+#if defined(__SIZEOF_INT128__)
+    #pragma GCC diagnostic push
+    #pragma GCC diagnostic ignored "-Wpedantic"
+    if (!is_constant_evaluated())
```
Comment (chfast) [inline include/intx/int128.hpp:368]: Using builtin uint128 is fine in constexpr too. The `is_constant_evaluated()` is only actually needed for MSVC usage of intrinsic. We can keep it if not too big problem.

**Comment by chfast** [comment]
> @chfast Not sure what to do with coverage here, any ideas?

Yes, the 32-bit will not have `__int128`. We just need 32-bit build + collect coverage from it.

**Comment by chfast** [comment]
> We just need 32-bit build + collect coverage from it.

This has landed, you can rebase now.

**Review by chfast — APPROVED** [review]

**Review by gumb0 — COMMENTED** [review]
Looks good to me.

**Pull request merged**